lighthouse-core/gather/gatherers/iframe-elements.js

```diff
@@ -0,0 +1,75 @@
+/**
+ * @license Copyright 2019 Google Inc. All Rights Reserved.
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with the License. You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the specific language governing permissions and limitations under the License.
+ */
+'use strict';
+
+const Gatherer = require('./gatherer.js');
+const pageFunctions = require('../../lib/page-functions.js');
+
+/* eslint-env browser, node */
+
+/**
+ * @return {LH.Artifacts['IFrameElements']}
+ */
+/* istanbul ignore next */
+function collectIFrameElements() {
+  // @ts-ignore - put into scope via stringification
+  const iFrameElements = getElementsInDocument('iframe'); // eslint-disable-line no-undef
+  return iFrameElements.map(/** @param {HTMLIFrameElement} node */ (node) => {
+    // @ts-ignore - put into scope via stringification
+    const clientRect = node.getBoundingClientRect().toJSON(); // eslint-disable-line no-undef
```

Comment:
this is about to go back over the protocol via JSON, so are the explicit `.toJSON()` and `// @ts-ignore` necessary?

Reply:
(and if it is necessary, need to remove the `eslint-disable-line` and update the `ts-ignore` comment. It's really that `getBoundingClientRect` will always return a `DOMRect` since this code only ever runs in Chrome, so `toJSON` will always be defined)

Reply:
I think those were leftover from when it was a helper function :)

the `toJSON` definitely used to be necessary because protocol didn't serialize the same way, but IIRC someone was saying it might be unnecessary these days? worth checking either way :)